Moodlet/app/src/main/java/com/cmput3owo1/moodlet/services/UserService.java

```diff
@@ -56,13 +58,12 @@ public boolean hasPreviousLogin(){
      */
     @Override
     public void validateUsernameAndCreateUser(final User user, final String password, final RegistrationListener listener){
-
+        listener.showProgressBar();
```

Comment:
I think it might be better to make this call outside of this function in the Fragments. Because now the UserService is doing UI actions when it just should be concerned with the DB actions. That way you don't need to add the showProgressBar to the listener.